SU2_CFD/src/solver_adjoint_discrete.cpp

```diff
@@ -479,7 +478,7 @@ void CDiscAdjSolver::ExtractAdjoint_Solution(CGeometry *geometry, CConfig *confi
 
   /*--- Set the old solution ---*/
 
-  nodes->Set_OldSolution();
+  if(!multizone) nodes->Set_OldSolution();
```

Comment:
This change is required for the zone-wise residuals to make sense, on entry Solution contains cross-terms and OF contributions but on extraction it does not, as the zone-wise residual is based on the before-after difference we were measuring the magnitude of the cross-terms and OF.
The dirty solution is to set OldSolution to the newly extracted one that does not contain those contributions yet.